CONTRIBUTING.md

```diff
@@ -35,24 +41,169 @@ You should receive a response within 24 hours. If for some reason you do not, pl
+1. Ensure spec is free of references, macros, comments, etc. only applicable to other distros
+1. Complete the [PR checklist](#pr-checklist)
+1. Pass the GitHub checks
+
+### Toolkit
+
+We welcome tooling improvements. When contributing to the toolkit, please adhere to `golang` formatting as described by the [fmt](https://pkg.go.dev/fmt) package. To format using this package, you can run `make go-tidy-all` in your CBL-Mariner toolkit. For guidance building with the toolkit, see our [building instructions](toolkit/docs/building/building.md).
+
+### Documentation
+
+We welcome documentation improvements. See [toolkit/docs](toolkit/docs) for the latest documentation.
+
+## Pull Request Guidelines
+
+Please direct pull requests to the desired development branch. Development changes to `2.0` should target `main`. Development changes to `1.0` should target `1.0-dev`.
+
+### Branch structure
+
+An overview of how the branches are structured can be seen below
+
+| Branch / Tag | For PRs | Published | Notes
+|:-------------|:--------|:----------|:------------
+|main          |Yes       |No         | **Primary development branch**
+|2.0           |No       |Yes - eventually   | Staging branch for publishing
+|2.0-stable    |No       |  Yes      | Last published release
+|2.0-preview   |No       |No         | Publishing in progress
+
+| Branch / Tag | For PRs | Published | Notes
+|:-------------|:--------|:----------|:------------
+|1.0-dev       |Yes      |No         | Development branch for 1.0
+|1.0           |No       |Yes - eventually    | Staging branch for publishing
+|1.0-stable    |No       |Yes        | Last published release
+|1.0-preview   |No       |No         | Publishing in progress
+
+### PR Titles
+
+PR titles should start with an action
+
+```bash
+- Add <package>
+- Bump Release version for October Update
+- Change whatever you changed.
+- Fix <thing you fixed> <reason you fixed it>
+- Patch <package> to fix CVE-XXXX-YYYY, CVE-XXXX-YYYY…
+- Upgrade <package> to version vvvv to fix CVE-XXXX-YYYY…
+- Remove <package> <reason you removed it>
+```
+
+Please avoid titles such as
+
+```bash
+- package: <whatever you did to the package>
+- CVE-XXXX-YYYY (leaving off what package was patched or upgraded)
+- [1.0] (prefixing with branch or other information)
+```
+
+### PR Checklist
+
+When creating your PR, please ensure the following:
+
+* The toolchain has been rebuilt successfully (or no changes were made to it). This only applies if you changed any packages in the [toolchain package manifests](toolkit/resources/manifests/package). Specifically, `toolchain_x86_64.txt` and `toolchain_aarch64.txt`. For guidance building the toolchain, see our [building instructions](toolkit/docs/building/building.md).
+
+* The [toolchain / worker package manifests](toolkit/resources/manifests/package) are up-to-date (versions match latest package versions in SPEC files). The manifests are the following files:
```

Comment:
I had issues here, i found that if i branched from main, the packages defined here were not always available in the upstream blob storage and i would get 404 when building the tool chain

Is it best practice before we build?
git checkout 2.0  -- toolchain_x86_64.txt

Reply:
`main` may include changes to the toolchain packages which have not yet been published. This is why they aren't available to be downladed. To avoid having to rebuild the toolchain, `2.0-stable` 's manifests can be used but just note that the manifests won't be guaranteed to match `main`. Therefore, when merging into `main` you still need to ensure you aren't rolling back TC versions